wherescape/wherescape.py

```diff
@@ -256,3 +289,105 @@ def write_parameter(self, name, value="", comment=None):
         result_number = self.push_to_meta(sql, [name, value, comment])
         result_number = int(result_number)
         return result_number
+
+    def job_clear_logs_by_date(self, days_to_retain=90, job_to_clean="%"):
+        """
+        Archives job logs that are older than the specified age in days.
+
+        Input:
+        ------
+        job_to_clean    : name of the job(s) to clean (Wild cards are supported)
+                            The name of the job(s) whose current logs are to be archived.
+                            Specifying % will match ALL jobs.
+        days_to_retain  : number of days to retain the logs for the specified job(s)
+                            If 90 days are retained then all the archived logs that are older than 90 days are
+                            archived. If 0 days are retained then all the logs are archived.
+        Returns:
+        return_code     : Output Return Code:
+                            • S - Success.
+                            • E - Error.
+                            • F - Fatal/Unexpected Error.
+        return_message  : Output message indicating the action applied or the reason for no action.
+        result_number:  : Output Result number:
+                            •  1 Success
+                            • -2 Error : Error. e.g. Due to invalid job name or job not running
+                            • -3 Fatal/Unexpected Error
+
+        """
+        function_parameter_list = [job_to_clean, days_to_retain]
+
+        sql = """
+        DECLARE @out nvarchar(max),@out1 nvarchar(max),@out2 nvarchar(max);
+        EXEC Ws_Job_Clear_Logs_By_Date
+         @p_sequence  =?
+        , @p_job_name  = ?
+        , @p_task_name  = ?
+        , @p_job_id = ?
+        , @p_task_id = ?
+        , @p_job_to_clean = ?
+        , @p_day_count = ?
+        , @p_return_code = @out OUTPUT
+        , @p_return_msg = @out1 OUTPUT
+        , @p_result   = @out2 OUTPUT;
+        SELECT @out AS return_code,@out1 AS return_msg,@out2 AS return_result"""
+
+        return_values = self.query_meta(
+            sql, self.common_input_parameter_list + function_parameter_list
+        )
+        return_code = return_values[0][0]
+        return_message = return_values[0][1]
+        result_number = int(return_values[0][2])
+
+        return return_code, return_message, result_number
+
+    def job_clear_archive_by_date(self, days_to_retain=365, job_to_clean="%"):
```

Comment:
❌
I'm not seeing the benefits. I think there is value in keeping the APIs similar to the existing WhereScape APIs. The added benefit would be very small, since the function descriptions will stay the same length, the sql-EXEC-strings will stay this same length, and the since the function parameter list is different, only the the last 4 lines of the functions are the same.